app/front/scripts/services/download-package.js

```diff
@@ -80,8 +80,8 @@
               StorageService.clearApplicationState()
                   .then(function() {
                     var packageName = PackageService.getAttributes().name;
-                    var owner = LoginService.email;
-                    state.packagePublicUrl = '/viewer/'+owner+':'+packageName; //dataPackage.uploadUrl;
+                    var owner = LoginService.userid;
```

Comment:
Well, based on my experience implementing this logic in a few systems over the last years, users do tend to login to services using the same email. So, if I have these accounts:

* Google
* Facebook
* GitHub
* BitBucket

There is a very high chance that I use the same email for all of them, that is why some auth implementations do this reconciliation based on email to associate different auth accounts with one user.

Also note that many of these services all the user to have multiple emails, that is why in the code snippet I link to above from the census, we store multiple email addresses for a user account (whatever emails the backend gives us) and we reconcile based on any of them